solutions/usaco-919.mdx

```diff
@@ -54,47 +59,40 @@ int main() {
 import java.util.*;
 import java.io.*;
 
-public class paintbarn {
-	static Kattio io;
-	static {
-		try {
-			io = new Kattio("paintbarn");
-		} catch(IOException e) {}
-	}
-
-	static int N, K, A=0;
-	static int[][] psum;
+public class PaintBarn {
+	private static final int WIDTH = 1000;
 	public static void main(String[] args) throws IOException {
-		N = io.nextInt();
-		K = io.nextInt();
-		psum = new int[1001][1001];
-
-		for (int i = 0; i < N; i++) {
-			int a, b, c, d;
-			a = io.nextInt()+1;
-			b = io.nextInt()+1;
-			c = io.nextInt()+1;
-			d = io.nextInt()+1;
-
-			// Setting up the prefix sums array with all the corners of the given rectangle
-			psum[a][b]++;
-			psum[c][d]++;
-			psum[a][d]--;
-			psum[c][b]--;
+		Kattio io = new Kattio("paintbarn");
+		int rectNum = io.nextInt();
+		int paintReq = io.nextInt();
+		int[][] barn = new int[WIDTH + 1][WIDTH + 1];
```

Comment:
why are we doing this? do dynamic sizing

Reply:
how- the barn's width is explicitly specified to be 1000

Reply:
wait mb i remebered the question wrong